#include "mainwindow.h"
#include "./ui_mainwindow.h"
#include "databasehandler.h"
#include <QSqlTableModel>
#include <QDebug>

#include "nuevoevento.h"

MainWindow::MainWindow(QWidget *parent)
    : QMainWindow(parent)
    , ui(new Ui::MainWindow)
{
    ui->setupUi(this); //carga interfaz visual

    //verificar conexiÃ³n a la base de datos
    if(!DatabaseHandler::instance()->openDatabase()){
        qDebug() << "Error: No se pudo conectar a la base de datos";
        return; //si falla, detiene aquÃ­
    }

    setupTableView();
}

void MainWindow::setupTableView()
{
    // ðŸ“Š 1. CONFIGURAR TABLA DE EVENTOS
    QSqlTableModel *modelEventos = new QSqlTableModel(this, DatabaseHandler::instance()->database());
    // ðŸŽ¯ Crear modelo que conecta la tabla SQL con la vista visual
    // ðŸ’¡ 'this' significa que se destruirÃ¡ automÃ¡ticamente cuando MainWindow se cierre

    modelEventos->setTable("eventos");
    // ðŸŽ¯ Especificar quÃ© tabla de la base de datos usar

    modelEventos->select();
    // ðŸš€ Cargar los datos de la tabla "eventos" desde la base de datos

    ui->tableViewEventos->setModel(modelEventos);
    // ðŸŽ¯ Vincular el modelo de datos con la tabla visual en la interfaz


    // ðŸ“Š 2. CONFIGURAR TABLA DE FALLAS BCH (misma lÃ³gica)
    QSqlTableModel *modelFallasBCH = new QSqlTableModel(this, DatabaseHandler::instance()->database());
    modelFallasBCH->setTable("FALLA_BCH");  // ðŸŽ¯ Tu segunda tabla
    modelFallasBCH->select();
    ui->tableViewFallasBCH->setModel(modelFallasBCH);


    // ðŸ“Š 3. CONFIGURAR TABLA DE FALLAS PW
    QSqlTableModel *modelFallasPW = new QSqlTableModel(this, DatabaseHandler::instance()->database());
    modelFallasPW->setTable("FALLA_PW");  // ðŸŽ¯ Tu tercera tabla
    modelFallasPW->select();
    ui->tableViewFallasPW->setModel(modelFallasPW);

    // // ðŸ“Š 1. CONFIGURAR TABLA DE EVENTOS
     QSqlTableModel *modelBCHRep = new QSqlTableModel(this, DatabaseHandler::instance()->database());
     modelBCHRep->setTable("REPARACION_BCH");
     modelBCHRep->select();
     ui->tableViewBCHR->setModel(modelBCHRep);

     QSqlTableModel *modelPWRep = new QSqlTableModel(this, DatabaseHandler::instance()->database());
     modelPWRep->setTable("REPARACION_PW");
     modelPWRep->select();
     ui->tableViewPWR->setModel(modelPWRep);


}

MainWindow::~MainWindow()
{
    delete ui;
}

void MainWindow::on_pushButton_NuevoEvento_clicked()
{
    NuevoEvento form(this);
    form.exec();
}

